src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -47,7 +48,10 @@ object RuleUtils {
           .signature(plan)
         signatureMap.put(sourcePlanSignature.provider, signature)
```

Comment:
can we use `getOrElseUpdate` to remove double look up?

Reply:
nice, added it.